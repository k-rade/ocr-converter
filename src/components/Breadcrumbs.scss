.breadcrumbs {

    display: flex;
    align-items: center;
    gap: 0.5rem;

    .breadcrumb {
        color: var(--color-white-two);

        &-divider {
            color: var(--color-gray-900) !important;
        }

        &-active {
            color: var(--color-primary-500);
            text-transform: capitalize;
        }
    }
}