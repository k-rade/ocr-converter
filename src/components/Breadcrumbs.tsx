import { NavLink, useLocation } from "react-router-dom";

import "./Breadcrumbs.scss";

const Breadcrumbs = (): JSX.Element => {
  const { pathname } = useLocation();

  return (
    <div className="breadcrumbs">
      <NavLink
        to="/"
        className={({ isActive }) =>
          `breadcrumb ${isActive ? "breadcrumb-active" : ""}`
        }
      >
        Home
      </NavLink>
      <span className="breadcrumb-divider">{">"}</span>
      <NavLink
        to={pathname}
        className={({ isActive }) =>
          `breadcrumb ${isActive ? "breadcrumb-active" : ""}`
        }
      >
        {pathname.replaceAll("/", "").replaceAll("-", " ")}
      </NavLink>
    </div>
  );
};

export default Breadcrumbs;
