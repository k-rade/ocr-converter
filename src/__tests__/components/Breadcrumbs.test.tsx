import { fireEvent, render } from "@testing-library/react";
import { MemoryRouter, useLocation } from "react-router-dom";
import Breadcrumbs from "../../components/Breadcrumbs";

jest.mock("react-router-dom", (): any => {
  const actualImplementation = jest.requireActual("react-router-dom");
  return {
    ...actualImplementation,
    useLocation: jest.fn(),
  };
});

describe("Breadcrumbs", (): void => {
  it("should render", (): void => {
    (useLocation as jest.Mock).mockImplementation(() => {
      return { pathname: "/test-route" };
    });

    const { baseElement } = render(
      <MemoryRouter>
        <Breadcrumbs />
      </MemoryRouter>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render with second route active", (): void => {
    (useLocation as jest.Mock).mockImplementation(() => {
      return { pathname: "/test-route" };
    });

    const { baseElement, getByText } = render(
      <MemoryRouter>
        <Breadcrumbs />
      </MemoryRouter>
    );

    fireEvent.click(getByText("test route"));

    expect(baseElement).toMatchSnapshot();
  });
});
