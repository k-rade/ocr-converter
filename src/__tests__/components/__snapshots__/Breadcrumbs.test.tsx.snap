// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render 1`] = `
<body>
  <div>
    <div
      class="breadcrumbs"
    >
      <a
        aria-current="page"
        class="breadcrumb breadcrumb-active"
        href="/"
      >
        Home
      </a>
      <span
        class="breadcrumb-divider"
      >
        &gt;
      </span>
      <a
        class="breadcrumb "
        href="/test-route"
      >
        test route
      </a>
    </div>
  </div>
</body>
`;

exports[`Breadcrumbs should render with second route active 1`] = `
<body>
  <div>
    <div
      class="breadcrumbs"
    >
      <a
        class="breadcrumb "
        href="/"
      >
        Home
      </a>
      <span
        class="breadcrumb-divider"
      >
        &gt;
      </span>
      <a
        aria-current="page"
        class="breadcrumb breadcrumb-active"
        href="/test-route"
      >
        test route
      </a>
    </div>
  </div>
</body>
`;
